File: 0509-fibonacci-number/0509-fibonacci-number.c


int fib(int n){
    if(n == 0)
        return 0;
    else if(n == 1)
        return 1;
    else
    {
        int first = 0;
        int second = 1;
        int next = 0;
        for(int i=0; i<n-1; i++)
        {
            next = first+second;
            first = second;
            second = next;
        }
        return next;
    }

}

File: 0509-fibonacci-number/0509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        if n == 0:
            return 0
        if n == 1:
            return 1
        first = 0
        second = 1
        new = 0
        for i in range(n-1):
            new = first+second
            first = second
            second = new
        return new
            
            
        